lib/manager/components/transform/AddCustomFile.js

```diff
@@ -0,0 +1,85 @@
+// @flow
+
+import React, { Component } from 'react'
+
+import type { AddCustomFileProps, ReplaceFileFromStringFields, TransformProps } from '../../../types'
+
+import CustomCSVForm from './CustomCSVForm'
+
+/**
+ * Component that renders fields for AddCustomFile. This transformation shares csvData props with the ReplaceFileFromString transformation.
+ * TODO: adapt this transformation to include a file upload for larger custom files?
+ */
+export default class AddCustomFile extends Component<TransformProps<AddCustomFileProps>, AddCustomFileProps> {
+  constructor (props: TransformProps<AddCustomFileProps>) {
+    super(props)
+    this.state = {csvData: props.transformation.csvData, table: props.transformation.table}
+  }
+
+  // $FlowFixMe: Flow doesn't recognize return type for these arrow functions
+  _handleChange = (evt: SyntheticInputEvent<HTMLInputElement>) => {
+    this.setState({...this.state, table: evt.target.value})
+  }
+
+  // $FlowFixMe[signature-verification-failure]
+  _onSaveCsvData = () => {
+    const csvData = this.state.csvData || null
+    const table = this.state.table || null
+    this.props.onSave({csvData, table}, this.props.index)
+  }
+
+  // $FlowFixMe[signature-verification-failure]
+  _onChangeCsvData = (evt: SyntheticInputEvent<HTMLInputElement>) => {
+    const newState = {...this.state, csvData: evt.target.value}
+    this.setState(newState)
+    this._updateErrors(newState)
+  }
+
+  _getValidationErrors (fields: ReplaceFileFromStringFields): Array<string> {
+    const issues = []
+    const { csvData } = fields
+
+    // CSV data must be defined.
+    if (!csvData || csvData.length === 0) {
+      issues.push('CSV data must be defined.')
+    } else if (!this.state.table) {
+      issues.push('Custom CSV must have a name.')
+    }
+    return issues
+  }
+
+  /**
+   * Notify containing component of the resulting validation errors if any.
+   * @param fields: The updated state. If not set, the component state will be used.
+   */
+  // $FlowFixMe[signature-verification-failure]
+  _updateErrors = (fields?: ReplaceFileFromStringFields) => {
+    const { onValidationErrors } = this.props
+    onValidationErrors(this._getValidationErrors(fields || this.state))
+  }
+
+  render () {
+    const {transformation} = this.props
+    const {csvData, table} = this.state
+    const inputIsUnchanged = csvData === transformation.csvData
```

Comment:
For clarity, this should probably be `inputIsChanged`

Reply:
Should it be changed? It is literally checking if the input is the same as the previous data and then is used like  `disabled={inputIsUnchanged}` which feels pretty intuitive to me. Lmk tho

Reply:
This introduces double negatives which we should avoid if possible

Reply:
It may introduce a double negative I suppose in the future if we negate that condition. Would you prefer `inputIsSame`?
If we moved to inputIsChanged, we introduce a negation which is effectively a degree more confusing than currently.